genie-agent/src/main/java/com/netflix/genie/agent/cli/JobRequestArgumentsImpl.java

```diff
@@ -162,6 +162,29 @@
     )
     private boolean jobRequestedViaAPI;
 
+    @Parameter(
+        names = {"--jobConfiguration"},
+        description = "URI or path of a job-level configuration file to attach, can be repeated",
```

Comment:
all of these say URI or path. If it's a path how are we handling it? We're just going to copy it into the working directory locally or are we going to convert it and push it somewhere in S3 first?

Reply:
You can see how it's handled in `ArgumentConverters.UriOrLocalPathConverter.class`, in short it's either a URI already, or i assume it's a local path and try to convert it to a URI (`file:///Users/foo/.../myfile.sql`).

So in the spec back from server, everything is a URI, and the resource loaded loads all of them regardless of where they come from (local disk, s3, ...)

I don't think we should be in the business of uploading to S3 to make a local file available remotely.
First of all, it's unnecessary.
Second, we are "sharing" a user file without the user realizing.